Yosemite/Yosemite/Stores/AccountCreationStore.swift

```diff
@@ -55,7 +53,41 @@ private extension AccountCreationStore {
                                                     password: password,
                                                     clientID: dotcomClientID,
                                                     clientSecret: dotcomClientSecret)
-            completion(result)
+            switch result {
+            case .failure(let error) where error == .invalidUsername:
```

Comment:
Hey @jaclync I was thinking that instead of retrying the request when we get an error due to invalid username for containing the blacklisted word `wordPress` in it, we could simply ensure when we load username suggestions that we remove any `wordPress` wording from the email. I think that'd make the code simpler and would save us from one additional API request. 
Wdyt?

Reply:
>I think that'd make the code simpler and would save us from one additional API request.

I hope that we still have to send a request to generate a user name after removing the `wordPress` word from the email. Please correct me if I am wrong.

Reply:
Oh, I understand. You meant to remove the `wordPress` keyword even before sending the first request for generating a username. 

I think it won't work if more keywords get added to the blacklist.

Reply:
@selanthiraiyan I think @JorgeMucientes meant that we remove any forbidden keyword like `wordpress` before generating username suggestions, so that it saves one API request to generate username suggestions again.

@JorgeMucientes That'd work for now, but I'm afraid that if there are more keywords added to the backend then the mobile client is stuck on the username error. I think this username error would be an edge case where an extra API request is needed, and Corey is planning to do a fix for this p1667234232298349/1666852701.302039-slack-C045CUK1Y3U.

Reply:
I don't have a strong opinion on either approach, so feel free to keep it as is 😄.  But to be honest, that blacklist doesn't seem to have changed ever, so it feels kind of YAGNI to anticipate a solution for a case that has never happened so far.  Anyway if Corey is going to take a look at patching this shortly I wouldn't worry too much about the solution used for this edge case.

Reply:
it's true that the forbidden list is most likely staying the same 😆 I'll go with this PR, since the root issue is going to be fixed shortly.